### ISSUE TITLE
Usage without attaching to specific elements

### ISSUE THREAD

**Issue by jkorrek** [nkappler/ctxmenu#27]
Hi there,

first of all: great work on this plugin!

I was looking for an alternative for our current context menu library an came across this library. It has almost everything I was looking for.

One suggestion / idea:

In our app, we are integrating a single context menu implementation with a couple of other controls ( tables, scheduling charts, maps, ...). 
In those cases it might not be possible or at least quite hard to setup a context menu for a specific target. Usually the libraries we use for the actual component already provide some sort of event handling for opening a custom context menu.

So for this to work, it would be great if your plugin would provide an additional api to show a context menu, based only on an already existing contextmenu event.

I did a simple implementation of this idea
https://github.com/jkorrek/ctxmenu/commit/18196b41f193284adc2a9fe76d1f3c2f27e70f2e
basically adding a `show` and `close` function to create (and close) a context menu "on demand" without attaching to specific targets.

Is this something you would be interested in for this plugin? If so I could clean up the code, extend the docs and open a pull request.

**Comment by nkappler** [comment]
Thanks for your feedback and your kind words 😊
I haven't had much time to look at your code, but I had a glance. If I understand you correctly, you want to add APIs to control the lifecycle of the menu instead of relying on events doing it for you, in other words, control the opening/closing programmatically, right?
I think there's no reason **not** to have this, in fact, I had some similar requests in the past, because of the `open` function, which is private at the moment.

Feel free to prepare a PR, I'll be happy to review it.
Please consider the following things:
- naming should be consistent: either `show` and `hide`, or `open` and `close`. I wouldn't want to mix it. I prefer show and hide and maybe we should rename the private `open` function to avoid future confusion
- The github pages file (./docs/demo.js) doubles as a manual test file, as there are no automated tests (yet). Feel free to add some buttons to the page to showcase the feature, or some hidden global functions.
- I've seen you've added tsdoc comments to the interface. If you could also add the documentation to the `README.md` that would be absolutely fantastic 💯 

Thank you!

**Comment by nkappler** [comment]
The private `open` method might not be required at all after your change, so if you can replace it with the new `show` method, feel free to delete it...

**Comment by jkorrek** [comment]
Great! I will give it go in the next couple of days.
